### PR TITLE
Keyframes documentation

### LINKED ISSUE #9625: Validator: allow additional amp-keyframes style
The goal is to allow a bigger `<style>` aimed for animations. Animations are often bigger in size due to heavy use of vendor prefixes and due to high detail in mutations. On the plus side, however, keyframes animations do not affect the critical rendering path.

The format will be:
```
<style amp-keyframes>
</style>
```

The following rules will be applied to this tag:
 1. The `amp-keyframes` style will only be allowed to contain `@keyframes` and `@media`/`@supports` with `@keyframes` rules and their vendor-prefixed versions.
 2. The maximum allowed size for this style tag is 200K
 3. Must be at the bottom of the document. In other words, it should come after all content elements.

Thus, a valid markup could look something like this:
```
<style amp-keyframes>
@keyframes anim1 {
  0% {transform: translateX(-100%);}
  100% {transform: translateX(100%);}
}
@-webkit-keyframes anim1 {
  0% {transform: translateX(-100%);}
  100% {transform: translateX(100%);}
}
@media (min-width: 300px) {
  @keyframes anim1 {
    0% {transform: translateX(-200%);}
    100% {transform: translateX(200%);}
  }
}
@supports (offset-distance: 0) {
  @keyframes anim1 {
    0% {offset-distance: 0}
    100% {offset-distance: 100%}
  }
}
</style>
```

### PR DESCRIPTION
**Pull request by dvoytenko**
Closes #9625.

/cc @alanorozco

### CODE REVIEW

File: spec/amp-html-format.md [hunk 1]
```diff
@@ -256,6 +256,8 @@ The following @-rules are allowed in stylesheets:
 
 Authors may add custom styles to a document using a single `<style amp-custom>` tag in the head of the document.
 
+In addition, a single `<style amp-keyframes>` tag can be used for `@keyframes` declarations in the end of the document. See [Keyframes stylesheet](#keyframes-stylesheet) section for more details.
```
Comment (bpaduch) [inline spec/amp-html-format.md]: As we currently allow `@keyframes` CSS declarations in the regular `<style amp-custom>` tag, why do we need this new style tag?  It'd be worthwhile to put that info in here.

If we're recommending to use keyframes in this new style tag, consider rewriting this area as follows: 

For `@keyframes` CSS declarations, add those rules to a `<style amp-keyframes>` tag, which must be located at the end of the AMP document. For details, see the [Keyframes stylesheet](#keyframes-stylesheet) section of this document.

File: spec/amp-html-format.md [hunk 2]
```diff
@@ -286,6 +288,28 @@ In the following examples `<property>` needs to be in the whitelist above.
 #### Maximum size
 It is a validation error if the author stylesheet is larger than 50,000 bytes.
 
+### Keyframes stylesheet
+
+In addition to `<style amp-custom>`, authors may also add `<style amp-keyframes>` tag.
```
Comment (bpaduch) [inline spec/amp-html-format.md]: Rewrite as follows:

In addition to the <style amp-custom> tag, authors may also add the <style amp-keyframes> tag, which is specifically for keyframes animations:

The following restrictions apply to the <style amp-keyframes> tag:

Comment (dvoytenko) [reply spec/amp-html-format.md]: Done

File: spec/amp-html-format.md (hunk 1 above)
Comment (dvoytenko) [reply spec/amp-html-format.md]: Rewrote according to your advise, but with some clarifications. Please PTAL again.

**Comment by dvoytenko** [comment]
@bpaduch PTAL

**Pull request merged**